core/txpool/legacypool/legacypool.go

```diff
@@ -683,6 +683,9 @@ func (pool *LegacyPool) validateTxBasics(tx *types.Transaction, local bool) erro
 		opts.MinTip = new(big.Int)
 	}
 	if err := txpool.ValidateTransaction(tx, pool.currentHead.Load(), pool.signer, opts); err != nil {
+		if pool.chainconfig.ChainID.Cmp(tx.ChainId()) != 0 {
+			log.Trace("Transaction targets wrong chain", "chainID", tx.ChainId(), "expected", pool.chainconfig.ChainID)
+		}
```

Comment:
I'm not sure if this is the right place to put this (I think we are trying to minimize the diff with geth in the core package.. Right? cc: @ceyonur)

If we do go with this, we shouldn't log this for unprotected transactions.
```suggestion
		if tx.Protected() && pool.chainconfig.ChainID.Cmp(tx.ChainId()) != 0 {
			log.Trace("Transaction targets wrong chain", "chainID", tx.ChainId(), "expected", pool.chainconfig.ChainID)
		}
```

Reply:
> If we do go with this, we shouldn't log this for unprotected transactions.


why?